lta/rest_server.py

```diff
@@ -767,53 +768,45 @@ class TransferRequestActionsPopHandler(BaseLTAHandler):
+        ret = []
         sdtr = self.db.TransferRequests
-        old_age = self.check_claims.old_age()
         query = {
-            "$and": [
-                {"source": {"$regex": f"^{src}"}},
-                {"$or": [
-                    {"claimed": False},
-                    {"claim_time": {"$lt": f"{old_age}"}}
-                ]}
-            ]
+            "source": source,
+            "status": "unclaimed",
         }
-        ret = []
         async for row in sdtr.find(filter=query,
-                                   projection=REMOVE_ID):
-            if (limit > 0):
-                uuid = row["uuid"]
-                row["claimant"] = pop_body
-                row["claimed"] = True
-                row["claim_time"] = now()
-                update_query = {
-                    "$and": [
-                        {"uuid": uuid},
-                        {"$or": [
-                            {"claimed": False},
-                            {"claim_time": {"$lt": f"{old_age}"}}
-                        ]}
-                    ]
-                }
-                update_doc = {"$set": row}
-                ret2 = await sdtr.find_one_and_update(filter=update_query,
-                                                      update=update_doc,
-                                                      projection=REMOVE_ID,
-                                                      return_document=AFTER)
-                if not ret2:
-                    logging.error(f"Unable to claim TransferRequest {uuid}")
-                else:
-                    logging.info(f"claimed TransferRequest {uuid} for {pop_body}")
-                    limit = limit - 1
-                    ret.append(row)
+                                   projection=REMOVE_ID,
+                                   limit=1,
+                                   sort=FIRST_IN_FIRST_OUT):
+            right_now = now()  # https://www.youtube.com/watch?v=nRGCZh5A8T4
+            uuid = row["uuid"]
+            row["status"] = "processing"
+            row["update_timestamp"] = right_now
+            row["claimed"] = True
+            row["claimant"] = claimant
+            row["claim_timestamp"] = right_now
+            update_query = {
+                "$and": [
+                    {"uuid": uuid},
+                    {"status": "unclaimed"},
+                ]
+            }
+            update_doc = {"$set": row}
+            ret2 = await sdtr.find_one_and_update(filter=update_query,
+                                                  update=update_doc,
+                                                  projection=REMOVE_ID,
+                                                  return_document=AFTER)
+            if not ret2:
+                logging.error(f"Unable to claim TransferRequest {uuid}")
+            else:
+                logging.info(f"claimed TransferRequest {uuid} for {claimant}")
+                ret.append(row)
```

Comment:
If we're only ever popping one transfer request, doing a `find` + `find_one_and_update` seems redundant.  Just do the single `find_one_and_update`.

Reply:
We could also not return a list, and just return the single TransferRequest.

Reply:
Latest update does it this way.